common/supplemental/languageInfo.xml

```diff
@@ -144,7 +144,10 @@ For terms of use, see http://www.unicode.org/copyright.html
 			<languageMatch desired="tt"	supported="ru"	distance="30"	oneway="true"/>	<!-- Tatar: tt ⇒ ru -->
 			<languageMatch desired="tum"	supported="en"	distance="30"	oneway="true"/>	<!-- Tumbuka: tum ⇒ en -->
 			<languageMatch desired="ug"	supported="zh"	distance="20"	oneway="true"/>	<!-- Uighur: ug ⇒ zh -->
-			<languageMatch desired="uk"	supported="ru"	distance="20"	oneway="true"/>	<!-- Ukrainian ⇒ Russian -->
+			
+			<languageMatch desired="uk"	supported="en"	distance="30"	oneway="true"/>	<!-- Ukrainian ⇒ English -->
+			<!-- CLDR-17382: languageMatch: Ukrainian should not fall back to Russian -->
```

Comment:
In that case then can we have the comment say that Ukrainian should match with English if we are adding an explicit match?

Reply:
I think it is ok to have the explicit fallback to en.

Reply:
I disagree very much. That's not how this is designed, nor advertised.